ab3d2_source/controlloop.s

```diff
@@ -473,6 +482,138 @@ READMAINMENU:
 
 				bsr		WAITREL
 				bra		.rdlop
+.nosave:
+***************************************************************
+;				cmp.w	#7,d0
+;				bne		playgame
+;				bsr		WAITREL
+
+;				bsr		customOptions
+
+;				lea		mnu_MYMAINMENU,a0
+;				bsr		MYOPENMENU
+
+				bsr		WAITREL
+				bra		.rdlop
+***************************************************************
+
+;fixme: there are better ways to do this, but it works.AL
+levelMenu:
+				lea		mnu_MYLEVELMENU,a0
+				bsr		MYOPENMENU
+
+				lea		mnu_MYLEVELMENU,a0
+				bsr		CHECKMENU
+
+				cmp.w	#8,d0
+				beq	levelMenu2
+				
+				cmp.w	#0,d0
+				bne.s	.sl2
+				move.w	#0,MAXLEVEL
+				bra	.levelSelectDone
+.sl2
+				cmp.w	#1,d0
+				bne.s	.sl3
+				move.w	#0,MAXLEVEL
```

Comment:
I don't know if I am missing something, but aren't all these equivalent to `move.w d0, MAXLEVEL` ? It seems like you could either range check d0 first, or even just mask it with $F and then move it to MAXLEVEL

Reply:
yeah, still learning.